lib/rules/empty-line-between-blocks.js

```diff
@@ -2,34 +2,96 @@
 
 var helpers = require('../helpers');
 
+var counter,
+    syntax;
+
 var findNearestReturn = function (parent, i) {
   var previous,
       doublePrevious,
       space;
 
-  if (i >= 2) {
-    doublePrevious = parent.content[i - 2];
+  if (syntax === 'scss') {
```

Comment:
Seen as you're already performing this checking in the rule itself would it not be better to split these into 2 separate functions? I feel this one is doing too much on it's own.